### DISCUSSION TITLE
List the dashboards used by that data source on its configuration page

### DISCUSSION THREAD

**Discussion by hedss** [grafana/grafana#46748] [Data Sources]
I've been doing a lot of work in the last few days with data sources in the demo kit.

Something has struck me, which is that I'd love to be able to know exactly which dashboards are currently using a particular data source. This would ideally be a new tab for each data source which lists anything that's referencing it directly by name or UID.

This would help determine what can be pruned, and what might need updating/testing should I change the configuration of that source.

Ideally, we'd also update the API so that I could get a list of specific dashboard UIDs that are using a data source I'm interested in (we could do this now getting a DS UID and then trawling through every single dashboard, but that's difficult when you have 10k dashboards!).

Discuss! 😁

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
